docs/examples/substrate/call_flags.sol

```diff
@@ -0,0 +1,23 @@
+contract CallFlags {
+
+    enum CallFlag { FORWARD_INPUT, CLONE_INPUT, TAIL_CALL, ALLOW_REENTRY }
+
+    // Set all flags found in _flags to true.
+    function bitflags(CallFlag[] _flags) internal pure returns (uint32 flags) {
+        for (uint n = 0; n < _flags.length; n++) {
+            flags |= (2 ** uint32(_flags[n]));
```

Comment:
Isn't is cheaper to do `2 << (uint32(_flags[n]) - 1)`?

Reply:
Might well be 🧠  gonna keep this in mind for a library version, thanks

Reply:
Ah, there is an even better choice: `1 << uint32(_flags[n])`.
Now power and no subtraction!

Reply:
Ah yes of course, left shift is equal to 2 to the power of it in that case, why didn't I think of this :sweat_smile: